bin/p2-pcctl/main.go

```diff
@@ -0,0 +1,122 @@
+	createAnnotations = cmdCreate.Flag("annotations", "Complete set of annotations - must parse as JSON!").String()
+
+	cmdGet   = kingpin.Command(CmdGet, "Show a pod cluster. ")
+	getPodID = cmdGet.Flag("pod", "The pod ID on the pod cluster").Required().String()
+	getAZ    = cmdGet.Flag("az", "The availability zone of the pod cluster").Required().String()
+	getName  = cmdGet.Flag("name", "The cluster name (ie. staging, production)").Required().String()
+
+	cmdDelete   = kingpin.Command(CmdDelete, "Show a pod cluster. ")
+	deletePodID = cmdDelete.Flag("pod", "The pod ID on the pod cluster").Required().String()
+	deleteAZ    = cmdDelete.Flag("az", "The availability zone of the pod cluster").Required().String()
+	deleteName  = cmdDelete.Flag("name", "The cluster name (ie. staging, production)").Required().String()
+)
+
+func main() {
+	cmd, consulOpts := flags.ParseWithConsulOptions()
+	client := kp.NewConsulClient(consulOpts)
+	kv := kp.NewConsulStore(client)
+	logger := logging.NewLogger(logrus.Fields{})
+	pcstore := pcstore.NewConsul(client, 3, &logger)
+
+	switch cmd {
+	case CmdCreate:
+		az := fields.AvailabilityZone(*createAZ)
+		cn := fields.ClusterName(*createName)
+		podID := types.PodID(*createPodID)
+		selector := selectorFrom(az, cn, podID)
+		pccontrol := control.NewPodCluster(az, cn, podID, pcstore, selector, kv.NewUnmanagedSession("pcctl", currentUserName()))
+
+		annotations := *createAnnotations
+		var parsedAnnotations map[string]interface{}
+		err := json.Unmarshal([]byte(annotations), &parsedAnnotations)
+		if err != nil {
+			log.Fatalf("could not parse json: %v", err)
+		}
+		_, err = pccontrol.Create(parsedAnnotations)
+		if err != nil {
+			log.Fatalf("err: %v", err)
+		}
+	case CmdGet:
+		az := fields.AvailabilityZone(*getAZ)
+		cn := fields.ClusterName(*getName)
+		podID := types.PodID(*getPodID)
+		selector := selectorFrom(az, cn, podID)
+
+		pccontrol := control.NewPodCluster(az, cn, podID, pcstore, selector, kv.NewUnmanagedSession("pcctl", currentUserName()))
+		pc, err := pccontrol.Get()
+		if err != nil {
+			fmt.Printf("Caught error while fetching pod cluster: %v", err)
+			os.Exit(1)
+		}
+
+		bytes, err := json.Marshal(pc)
+		if err != nil {
+			logger.WithError(err).Errorln("Unable to marshal PC as JSON")
+		}
+		fmt.Printf("%s", bytes)
+	case CmdDelete:
+		az := fields.AvailabilityZone(*deleteAZ)
+		cn := fields.ClusterName(*deleteName)
+		podID := types.PodID(*deletePodID)
+		selector := selectorFrom(az, cn, podID)
+		pccontrol := control.NewPodCluster(az, cn, podID, pcstore, selector, kv.NewUnmanagedSession("pcctl", currentUserName()))
+		errors := pccontrol.Delete()
+		if len(errors) > 1 {
+			for _, err := range errors {
+				_, _ = os.Stderr.Write([]byte(fmt.Sprintf("Failed to one pod cluster matching arguments. Error:\n %s\n", err.Error())))
+			}
```

Comment:
It'd be a desirable property to exit non-0 if there are any errors (so maybe keep the if statement above, just fix it)